pkg/networkservice/common/authorize/server.go

```diff
@@ -28,19 +35,69 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
-type authorizeServer struct{}
+type authorizeServer struct {
+	p opa.Provider
+}
 
 // NewServer - returns a new authorization networkservicemesh.NetworkServiceServers
-func NewServer() networkservice.NetworkServiceServer {
-	return &authorizeServer{}
+func NewServer(p opa.Provider) networkservice.NetworkServiceServer {
```

Comment:
Would it be possible to keep the amount of new API to a minimum and just pass in a func() rego.PreparedEvalQuery ?

Reply:
Sure, I'll roll back my changes. I misinterpreted Fred's comment about separate service for OPA. I thought we want to obtain policies from some service, but now it is clear that Fred was talking only about evaluation.

Reply:
Sorry about the confusion, I'll try to be more specific moving forward.